testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/settings/SettingsMenuBaseActivity.kt

```diff
@@ -42,25 +49,55 @@ abstract class SettingsMenuBaseActivity: BaseActivity() {
     }
 
     private fun renderAppSettingsDialog(
-        settingsDialog: View?,
+        settingsDialog: View,
         appId: EditText,
         subscriptionKey: EditText
     ) {
-        AlertDialog.Builder(this)
+        val dialog = AlertDialog.Builder(this)
             .setTitle(R.string.lb_app_settings)
             .setView(settingsDialog)
-            .setPositiveButton(R.string.action_save) { dialog, _ ->
-                dialog.dismiss()
-                updateSettings(appId.text.toString(), subscriptionKey.text.toString())
+            .setPositiveButton(R.string.action_save, null)
+            .setNegativeButton(android.R.string.cancel, null)
+            .create()
+
+        dialog.setOnShowListener { _dialog ->
+            (_dialog as AlertDialog).getButton(AlertDialog.BUTTON_POSITIVE).setOnClickListener {
+                val pb = settingsDialog.findViewById<View>(R.id.pb)
+                pb.visibility = View.VISIBLE
+                updateSettings(appId.text.toString(), subscriptionKey.text.toString(), pb)
+            }
+
+            _dialog.getButton(AlertDialog.BUTTON_NEGATIVE).setOnClickListener {
+                _dialog.dismiss()
             }
-            .setNegativeButton(android.R.string.cancel) { dialog, _ -> dialog.cancel() }
-            .show()
+        }
+
+        dialog.show()
     }
 
-    private fun updateSettings(appId: String, subscriptionKey: String) {
+
+    private fun updateSettings(appId: String, subscriptionKey: String, pb: View) {
+        val appIdHolder = settings.appId
+        val subscriptionKeyHolder = settings.subscriptionKey
         settings.appId = appId
         settings.subscriptionKey = subscriptionKey
 
-        recreate()
+        launch {
+            try {
+                MiniApp.instance(AppSettings.instance.miniAppSettings).listMiniApp()
+                runOnUiThread {
+                    pb.visibility = View.GONE
+                    recreate()
+                }
+            } catch (error: MiniAppSdkException) {
+                settings.appId = appIdHolder
+                settings.subscriptionKey = subscriptionKeyHolder
+                runOnUiThread {
+                    pb.visibility = View.GONE
+                    Toast.makeText(this@SettingsMenuBaseActivity, error.message, Toast.LENGTH_LONG)
```

Comment:
I think the toast message is difficult to see because it displays on top of the keyboard. Maybe you can change the position so it's displayed at the top of the screen?